registry-contracts/src/v2/AgreementV2.sol

```
@@ -60,96 +79,83 @@ contract AgreementV2 is Ownable {
     }
 
     function addChains(Chain[] memory _chains) external onlyOwner {
+        // Validate chain IDs are valid
+        _validateChainIds(_chains);
+        // Validate no duplicates with existing chains
         for (uint256 i = 0; i < _chains.length; i++) {
```

Don't you have `validateNoDuplicateChains` for this?

---

oh so you update it the state and then check? I think failing fast is better though no?